deepchecks/nlp/text_data.py

```diff
@@ -0,0 +1,424 @@
+
+    Parameters
+    ----------
+    raw_text : t.Sequence[str]
+        The raw text data, a sequence of strings representing the raw text of each sample.
+    label : t.Optional[TTextLabel], default: None
+        The label for the text data. Can be either a text_classification label or a token_classification label.
+        If None, the label is not set.
+        - text_classification label - For text classification the accepted label format differs between multilabel and
+          single label cases. For single label data, the label should be passed as a sequence of labels, with one entry
+          per sample that can be either a string or an integer. For multilabel data, the label should be passed as a
+          sequence of sequences, with the sequence for each sample being a binary vector, representing the presence of
+          the i-th label in that sample.
+        - token_classification label - For token classification the accepted label format is a sequence of sequences,
+          with the inner sequence containing tuples in the following format: (class_name, span_start, span_end).
+          span_start and span_end are the start and end character indices of the token within the text, as it was
+          passed to the raw_text argument. Each upper level sequence contains a sequence of tokens for each sample.
+    task_type : str, default: None
+        The task type for the text data. Can be either 'text_classification' or 'token_classification'. Must be set if
+        label is provided.
+    classes:  t.Optional[t.Sequence[str]], default: None
+        The class names for the multilabel text classification task. May be set to define names for the classes in
+        multilabel tasks, as the label input is a binary matrix that cannot convey the class names. Length must match
+        the number of classes in the label, which is the number of columns for multilabel labels. Order of classes in
+        this input must match order of classes as returned by model predictions to ensure they are presented correctly.
+    dataset_name : t.Optional[str], default: None
+        The name of the dataset. If None, the dataset name will be defined when running it within a check.
+    index: t.Optional[t.Sequence[int]], default: None
+        The index of the samples. If None, the index is set to np.arange(len(raw_text)).
+    """
+
+    _text: t.Sequence[str]
+    _label: TTextLabel
+    _index: t.Sequence[int]
+    _task_type: t.Optional[TaskType]
+    _has_label: bool
+    _is_multilabel: bool = False
+    _classes: t.Optional[t.List[t.Union[str, int]]] = None
+    name: t.Optional[str] = None
+
+    def __init__(
+            self,
+            raw_text: t.Sequence[str],
+            label: t.Optional[TTextLabel] = None,
+            task_type: str = None,
+            classes: t.Optional[t.Sequence[str]] = None,
+            dataset_name: t.Optional[str] = None,
+            index: t.Optional[t.Sequence[int]] = None,
```

Comment:
I think index is confusing, why is it important to get the original indexes of user?
We just need to validate len(label) == len(raw_text) and assume their order match. 
We do the same in tabular

Reply:
In tabular we do accept the user indexes for the index related checks.
I do admit though that I need this argument also for creating sampled versions of a dataset, and if we remove it I'll need to think how to handle it otherwise.

Reply:
Oh, i see. as long as it optional I can live with it